src/main/java/com/gd/clinic/controller/PatientsController.java

```diff
@@ -4,33 +4,39 @@
 import com.gd.clinic.model.NewPatientDto;
 import com.gd.clinic.model.PatientResponseDto;
 import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.RestController;
 
 import java.util.List;
 
 @RestController
 public class PatientsController implements PatientsApi {
 
+    @PreAuthorize("hasRole('DOCTOR')")
     @Override
     public ResponseEntity<PatientResponseDto> addPatient(NewPatientDto newPatientDto) {
```

Comment:
if you are referring to the addPatient method. I tried to generalize the name like you said, but since the method is being implemented from the PatientsApi it doesnt allow the same name to be repeated. I modified it the api.yml to test and it gives error saying that the operationId is repeated if I give the same simple name to the others